src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1948,14 +1948,12 @@ protected void EnsureNullabilityAnalysisPerformedIfNecessary()
 
             upgradeableLock.EnterWrite();
 
-            Debug.Assert(Root == bindableRoot);
```

Comment:
Please test IOperation behavior for when nullable is enabled and the `unchecked`/`unsafe` block is the root. In particular, what the behavior of calling `SemanticModel.GetIOperation` on that `UnsafeStatementSyntax` is. The `Type` property of those nodes will be wrong unless you wait for [https://github.com/dotnet/roslyn/pull/42403](https://github.com/dotnet/roslyn/pull/42403) to merge and merge 16.8-p1 into your local code, but I'd expect other symbols like the method symbols of `IInvocationOperation`s or local symbols to reflect the correct nullability.

Reply:
> In particular, what the behavior of calling SemanticModel.GetIOperation on that UnsafeStatementSyntax is.

I would expect to get nothing as for any node that disappears from the tree, parenthesized and such.

> The Type property of those nodes will be wrong unless you wait for [https://github.com/dotnet/roslyn/pull/42403](https://github.com/dotnet/roslyn/pull/42403) to merge and merge 16.8-p1 into your local code, but I'd expect other symbols like the method symbols of IInvocationOperations or local symbols to reflect the correct nullability.

These are statements, they do not have types, and I expect to get no IOperation node for them.


---
In reply to: [444547469](https://github.com/dotnet/roslyn/pull/45334#discussion_r444547469) [](ancestors = 444547469)